### PR TITLE
docs: Update Release_Management.md

### PR DESCRIPTION
**Pull request by tam7t**
Match branch naming style

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/Release_Management.md [hunk 1]
```diff
@@ -30,20 +29,15 @@ This project strictly follows [semantic versioning](https://semver.org/spec/v2.0
...
 - Secrets Store CSI Driver is currently tracking towards first stable release(v1.0.0) with [this](https://github.com/kubernetes-sigs/secrets-store-csi-driver/milestone/5) milestone.
 
-
-
 ## Release Cadence and Branching
 - Secrets Store CSI Driver follows `monthly` release schedule.
 
 - A new release should be created in _`second week`_ of each month. This schedule not only allows us to do bug fixes, but also provides an opportunity to address underline image vulnerabilities etc. if any.
 
-- The release candidate (RC) images will be published from the master branch with tags. Once we validate RC image, we'll cut a release branch.
```
Comment (ritazh) [inline docs/Release_Management.md:40]: Are we ever going to cut RCs?

Comment (tam7t) [reply docs/Release_Management.md:40]: Seems reasonable to do RCs but I wanted to make this match the process that we're actually following in https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/docs/RELEASE.md

I also opened https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/688 to document some issues in our image promotion process.

Comment (ritazh) [reply docs/Release_Management.md:40]: I see. I guess we can add RC steps back when we start to cut RC. WDYT?

Comment (tam7t) [reply docs/Release_Management.md:40]: Sorry just got back to this - Yeah I think if we want to do RC releases then we'll want / need to fully document that process (and the process for promoting the RC to release tags).

Comment (aramase) [reply docs/Release_Management.md:40]: Documenting the process for RC when we start doing it sgtm!

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**